lib/active_admin_sidebar/positions.rb

```diff
@@ -1,10 +1,25 @@
 module ActiveAdminSidebar
   module Positions
-    def left_sidebar!
-        @sidebar_position = :left
+
+    def left_sidebar!(options={})
+        @sidebar_options = options.merge(position: :left)
+        if @sidebar_options.fetch(:collapsed, false)
+          collapsed_sidebar
+          @sidebar_options.merge!(sidebar_is_collapsed: session[:collapsed_sidebar])
+        end
     end
+
     def right_sidebar!
-        @sidebar_position = :right
+      @sidebar_options = {position: :right}
+    end
+
+    def collapsed_sidebar
+      if request.xhr?
+        if params[:collapsed_sidebar].present?
+          session[:collapsed_sidebar] = (params[:collapsed_sidebar] == 'true')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.